pkg/policy/selectorcache.go

```diff
@@ -477,50 +583,84 @@ func (sc *SelectorCache) UpdateIdentities(added, deleted cache.IdentityCache) {
 		}
 	}
 	for numericID, lbls := range added {
-		if _, exists := sc.idCache[numericID]; exists {
-			log.WithFields(logrus.Fields{logfields.Identity: numericID}).Warning("UpdateIdentities: Adding an existing identity")
+		if old, exists := sc.idCache[numericID]; exists {
+			// Skip if no change
+			if lbls.String() == old.lbls.String() {
+				log.WithFields(logrus.Fields{logfields.Identity: numericID}).Debug("UpdateIdentities: Skipping add an existing identical identity")
+				delete(added, numericID)
+				continue
+			}
+			log.WithFields(logrus.Fields{logfields.Identity: numericID, logfields.Labels: old.lbls, logfields.Labels + "(new)": lbls}).Warning("UpdateIdentities: Updating an existing identity")
+		} else {
+			log.WithFields(logrus.Fields{logfields.Identity: numericID, logfields.Labels: lbls}).Debug("UpdateIdentities: Adding a new identity")
 		}
 		sc.idCache[numericID] = newIdentity(numericID, lbls)
 	}
 
-	// Iterate through all locally used identity selectors and
-	// update the cached numeric identities as required.
-	for _, sel := range sc.selectors {
-		var adds, dels []identity.NumericIdentity
-		switch idSel := sel.(type) {
-		case *labelIdentitySelector:
-			for numericID := range deleted {
-				if _, exists := idSel.cachedSelections[numericID]; exists {
-					dels = append(dels, numericID)
-					delete(idSel.cachedSelections, numericID)
+	if len(deleted)+len(added) > 0 {
+		sc.idCacheRevision++
+
+		// Iterate through all locally used identity selectors and
+		// update the cached numeric identities as required.
+		for _, sel := range sc.selectors {
+			var adds, dels []identity.NumericIdentity
+			switch idSel := sel.(type) {
+			case *labelIdentitySelector:
+				for numericID := range deleted {
+					if _, exists := idSel.cachedSelections[numericID]; exists {
+						dels = append(dels, numericID)
+						delete(idSel.cachedSelections, numericID)
+					}
 				}
-			}
-			for numericID := range added {
-				if _, exists := idSel.cachedSelections[numericID]; !exists {
-					if idSel.Matches(sc.idCache[numericID]) {
-						adds = append(adds, numericID)
-						idSel.cachedSelections[numericID] = struct{}{}
+				for numericID := range added {
+					if _, exists := idSel.cachedSelections[numericID]; !exists {
+						if idSel.matches(sc.idCache[numericID]) {
+							adds = append(adds, numericID)
+							idSel.cachedSelections[numericID] = struct{}{}
+						}
 					}
 				}
+				if len(dels)+len(adds) > 0 {
+					idSel.updateSelections()
+					idSel.notifyUsers(adds, dels)
+				}
+			case *fqdnSelector:
+				// This is a no-op right now. We don't encode in the identities
+				// which FQDNs they correspond to.
 			}
-			if len(dels)+len(adds) > 0 {
-				idSel.updateSelections()
-				idSel.notifyUsers(adds, dels)
-			}
-		case *fqdnSelector:
-			// This is a no-op right now. We don't encode in the identities
-			// which FQDNs they correspond to.
 		}
 	}
 }
 
+// XXXGetIDCacheRevision can be used to figure our if the selections
```

Comment:
Why XXX?

our --> out

Reply:
This should not be necessary when we move to incremental updates. "XXX" to remind that this is supposed to be a temporary solution.